dokka-runners/dokka-gradle-plugin/src/main/kotlin/formats/DokkaPublication.kt

```diff
@@ -38,30 +46,94 @@ constructor(
+     * For example, by not resolving class/member links from your dependencies, including the standard library.
+     *
+     * Note: You can cache files locally and provide them to Dokka as local paths.
+     * See [DokkaSourceSetSpec.externalDocumentationLinks].
+     */
     abstract val offlineMode: Property<Boolean>
 
+    /**
+     * Whether to fail documentation generation if Dokka has emitted a warning or an error.
+     * The process waits until all errors and warnings have been emitted first.
+     *
+     * This setting works well with [DokkaSourceSetSpec.reportUndocumented].
+     */
     abstract val failOnWarning: Property<Boolean>
 
+    /**
+     * Whether to suppress obvious functions.
+     *
+     * A function is considered to be obvious if it is:
+     *
+     * - Inherited from [kotlin.Any], [kotlin.Enum], [java.lang.Object] or [java.lang.Enum],
+     *   such as `equals()`, `hashCode()`, `toString()` functions.
+     * - Synthetic (generated by the compiler) and does not have any documentation,
+     *   such as `data class` `componentN()` or `copy()` functions.
+     */
     abstract val suppressObviousFunctions: Property<Boolean>
 
+    /**
+     * Whether to suppress inherited members that aren't explicitly overridden in a given class.
+     *
+     * Note: This can suppress functions such as `equals()`, `hashCode()`, or `toString()`,
+     * but cannot suppress synthetic functions such as `data class` `componentN()` or `copy()`.
+     * Instead, use [suppressObviousFunctions].
+     */
+    abstract val suppressInheritedMembers: Property<Boolean>
+
+    /**
+     * A list of Markdown files that contain module and package documentation.
+     *
+     * The contents of the specified files are parsed and embedded into documentation as module and package descriptions.
+     *
+     * The format of the Markdown files is very specific and must be exactly correct to be readable by Dokka.
+     *
+     * See the
+     * [Dokka Gradle example](https://github.com/Kotlin/dokka/tree/master/examples/gradle-v2/basic-gradle-example)
+     * for an example of what it looks like and how to use it.
+     */
     abstract val includes: ConfigurableFileCollection
 
-    abstract val suppressInheritedMembers: Property<Boolean>
+    /**
+     * A shared cache directory used by DokkaGenerator.
+     *
+     * (Aside: This property is not used by any official Dokka plugins.
+     * It has been retained in case third-party Dokka plugins use it.)
+     */
+    abstract val cacheRoot: DirectoryProperty
 
-    // TODO probably not needed any more, since Dokka Generator now runs in an isolated JVM process
+    /**
+     * Controls whether [org.jetbrains.dokka.DokkaGenerator] will forcibly stop coroutines.
+     *
+     * This is an internal Dokka Gradle plugin property.
+     * If you find you need to set this property, please report your use-case https://kotl.in/dokka-issues.
+     */
```

Comment:
let's add `@DokkaInternalApi`?

Reply:
https://youtrack.jetbrains.com/issue/KT-72288/Require-an-opt-in-requirement-for-DokkaPublicationfinalizeCoroutines